addons/mail/models/mail_channel.py

```diff
@@ -805,8 +805,8 @@ def channel_info(self):
                     'seen_message_id': cp.seen_message_id.id,
                 } for cp in members_by_channel[channel] if cp.partner_id], key=lambda p: p['partner_id'])
                 info['guestMembers'] = [('insert', sorted([{
-                    'id': member.guest_id.id,
-                    'name': member.guest_id.name,
+                    'id': member.guest_id.sudo().id,
+                    'name': member.guest_id.sudo().name,
```

Comment:
Tests do not crash when I remove this `sudo`. Test coverage should be extended to cover it if it is necessary, or it should be reverted if not.